SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -40,13 +42,14 @@ Buffer {
 		bufBase = bufnum ?? { server.nextBufferNumber(numBufs) };
 		numFrames = numFrames.asInteger;
 		numChannels = numChannels.asInteger;
+		if(sampleRate.isNil or: {sampleRate <= 0}) { sampleRate = server.sampleRate };
 		^Array.fill(numBufs, { |i|
 			// note, cannot use alloc or allocMsg here
 			// because those methods don't pass a buffer index for completion message
 			var newBuf = Buffer.new(server, numFrames, numChannels, i + bufBase, sampleRate);
 			server.sendMsg(\b_alloc, i + bufBase, numFrames, numChannels,
 				completionMessage.value(newBuf, i),
-				sampleRate !? _.asFloat ?? 0.0);
+				sampleRate);
```

Comment:
we need sampleRate.asFloat here (for supernova, since we are sending the message directly)

Reply:
this still needs to be done, otherwise supernova complains for "wrong argument type"
```supercollider
server.sendMsg(\b_alloc, i + bufBase, numFrames, numChannels,
				completionMessage.value(newBuf, i),
				sampleRate.asFloat);
```